modules/core/include/opencv2/core/quaternion.hpp

```diff
@@ -0,0 +1,1949 @@
+         * @param assumeUnit if true, quaternion q should be a unit quaternion and this function will save some computations.
+         *
+         * For example
+         * ```
+         * Quatf q(1,2,3,4);
+         * sqrt(q);
+         *
+         * q = {1,0,0,0};
+         * sqrt(q, true); //true means q is a unit quaternion
+         * ```
+         */
+        template <typename T>
+        friend Quat<T> sqrt(const Quat<T> &q, bool assumeUnit);
+
+        /**
+         * @brief return \f$\sqrt{q}\f$
+         * @param assumeUnit if true, this quaternion should be a unit quaternion and this function will save some computations.
+         *
+         * For example
+         * ```
+         * Quatf q(1,2,3,4);
+         * q.sqrt();
+         *
+         * q = {1,0,0,0};
+         * q.sqrt(true); //true means q is a unit quaternion
+         * ```
+         */
+        Quat<_Tp> sqrt(bool assumeUnit=false) const;
+
+        /**
+         * @brief return the value of power function with quaternion \f$q\f$
+         * \f[p^q = e^{q\ln(p)}\f]
+         * @param p base quaternion of power function
+         * @param q index quaternion of power function
+         * @param assumeUnit if true, quaternon \f$p\f$ should be a unit quaternion and this function will save some computations
+         *
+         * For example
+         * ```
+         * Quatd p(1,2,3,4);
+         * Quatd q(5,6,7,8);
+         * power(p, q);
+         *
+         * p = p.normalize();
+         * power(p, q, true); //true means p is a unit quaternion
+         * ```
+         */
+        template <typename T>
+        friend Quat<T> power(const Quat<T> &p, const Quat<T> &q, bool assumeUnit);
+
+        /**
+         * @brief return the value of power function with quaternion \f$q\f$
+         * \f[p^q = e^{q\ln(p)}\f]
+         * @param q index quaternion of power function
+         * @param assumeUnit if true, this quaternon should be a unit quaternion and this function will save some computations
+         *
+         * For example
+         * ```
+         * Quatd p(1,2,3,4);
+         * Quatd q(5,6,7,8);
+         * p.power(q);
+         *
+         * p = p.normalize();
+         * p.power(q, true); //true means p is a unit quaternion
+         * ```
+         */
+        Quat<_Tp> power(const Quat<_Tp> &q, bool assumeUnit=false) const;
+
+        /**
+         * @brief return the crossProduct between \f$p = (a, b, c, d) = (a, \boldsymbol{u})\f$ and \f$q = (w, x, y, z) = (w, \boldsymbol{v})\f$
+         * \f[p \times q = \frac{pq- qp}{2}\f]
+         * \f[p \times q = \boldsymbol{u} \times \boldsymbol{v}\f]
+         * \f[p \times q = (cz-dy)i + (dx-bz)j + (by-xc)k \f]
+         *
+         * For example
+         * ```
+         * Quatd q{1,2,3,4};
+         * Quatd p{5,6,7,8};
+         * crossPruduct(p, q)
+         * ```
+         */
+        template <typename T>
+        friend Quat<T> crossProduct(const Quat<T> &p, const Quat<T> &q);
```

Comment:
It can be implemented without friendship, because there is public method  `crossProduct`.